mergebot_poller.py

```diff
@@ -1,47 +1,126 @@
+    """
+    # Instantiate the two variables used for tracking work.
+    # Thread-safe because of the GIL.
+    self.work_queue = Queue.Queue()
+    self.known_work = set()
+    # Set up a github helper for handling network requests, etc.
+    self.github_helper = GithubHelper(ORG, self.config['repository'])
+    # Kick off an SCM merger
+    merger = Thread(target = SCM_MERGERS['github'],
+                    args = (self.work_queue, self.known_work, self.github_helper, ))
+    merger.start()
+    # Loop: Forever, every fifteen seconds.
     while True:
-      print('Loading pull requests from Github at {}.'.format(PULLS_URL))
       # Load list of pull requests from Github
-      r = requests.get(PULLS_URL, auth=(BOT_NAME, bot_key))
-      if r.status_code != 200:
-        print('Oops, that didn\'t work. Error below, waiting then trying again.')
-        print(r.text)
-        return
-
-      print('Loaded.')
-      pr_json = r.json()
-      # Loop: Each pull request
-      for pr in pr_json:
-        search_pr(pr)
-
+      prs = self.github_helper.fetch_prs()
+      if prs is not None:
+        # Loop: Each pull request
+        for pr in prs:
+          if pr['number'] not in self.known_work:
+            search_github_pr(pr)
+      else:
+        print('Error fetching pull requests.')
       time.sleep(15)
 
+  def search_github_pr(self, pr):
+    """
+    Searches a provided github PR for mergebot commands, validates, and runs
+    said commands.
+
+    :param pr: The pull request to validate
+    :return: returns nothing
+    """
+    pr_num = pr['number']
+    print('Looking at PR #{}.'.format(pr_num))
+    # Load comments for each pull request
+    cmts = self.github_helper.fetch_comments(pr_num)
+    if len(cmts) < 1:
+      print('No comments on PR #{}. Moving on.'.format(pr_num))
+      return
+    # FUTURE: Loop over comments to make sure PR has been approved by a
+    # committer.
+    cmt = cmts[-1]
+    cmt_body = cmt['body'].encode('ascii', 'ignore')
+    # Check auth.
+    user = cmt['user']['login']
+    if user not in AUTHORIZED_USERS:
+      error = 'User {} not a committer; access denied.'.format(user)
+      self.github_helper.post_error(error)
+      print('Unauthorized user {} attempted command, #{}.'.format(user, pr_num))
+      return
+    # Look for committer request comment.
+    # FUTURE: Look for @merge-bot reply comments.
+    # FUTURE: Use mentions API instead?
+    if not cmt_body.startswith('@{}'.format(BOT_NAME)):
+      print('Last comment: {}, not a command. Moving on.'.format(cmt_body))
+      return
+    cmd_str = cmt_body.split('@{} '.format(BOT_NAME), 1)[1]
+    cmd = cmd_str.split(' ')[0]
+    if cmd not in CMDS.keys():
+      # Post back to PR
+      self.github_helper.post_error('Command was {}, not a valid command.'
+                                    ' Valid commands: {}.'.format(cmd, CMDS.keys()), pr_num)
+      print('Command was {}, not a valid command.'.format(cmd))
+      return
```

Comment:
(1) Because we're posting to the thread and we only use the last comment we won't keep complaining about a particular command.
(2) In the first implementation, there is only one command: merge. Any more complicated interpretation of comments is deferred to later implementations.